frontend/components/site/siteBuildLogs.jsx

```diff
@@ -1,23 +1,49 @@
 /* eslint-disable react/forbid-prop-types */
 
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link, useParams } from 'react-router-dom';
 
 import { useBuildLogs } from '../../hooks';
+import LoadingIndicator from '../LoadingIndicator';
 import SiteBuildLogTable from './siteBuildLogTable';
 import DownloadBuildLogsButton from './downloadBuildLogsButton';
 import CommitSummary from './CommitSummary';
+import api from '../../util/federalistApi';
 
 export const REFRESH_INTERVAL = 15 * 1000;
+const BUILD_LOG_RETENTION_LIMIT = 180 * 24 * 60 * 60 * 1000; // 180 days in milliseconds
+
+function buildTooOld(build) {
+  return (new Date() - new Date(build.completedAt)) > BUILD_LOG_RETENTION_LIMIT;
+}
 
 const SiteBuildLogs = () => {
   const { buildId: buildIdStr } = useParams();
   const buildId = parseInt(buildIdStr, 10);
   const { logs, state } = useBuildLogs(buildId);
 
+  const [build, setBuild] = useState({
+    isLoading: true,
+    buildDetails: null,
+  });
+  const { isLoading, buildDetails } = build;
+
+  useEffect(() => {
```

Comment:
Since we use this hook in two places, what do you think about creating a new build details hook in the `hooks` directory and importing it to the paces we use it?

Reply:
OK, I'll see what I can do.